### PR TITLE
MTM-46782 Adding a minified partial extract from the core model to de…

### PR DESCRIPTION
**Pull request by ghost**
Draft: This makes the packaging independent from core-model, by forking a (minified) model from core into package.

This raises some questions, for example if this model should be in an own maven module or not...

### CODE REVIEW

**Comment by torsSAG** [comment]
One simple question, is it a good choice to use svenson for implementing new stuff. At least GSON or Jackson so far, svenson was implemented for using Java 5 and there is no work done for years now. It could maybe cause problems if we switch to newer JDK versions.

**Comment by kaczynskid-sag** [comment]
That's no new stuff @torsSAG, Svenson was there all the time. What Elias did was just a copy of the code that was previously referenced as a compile time dependency from core. This is a valid first step and replacing the object mapper might be a valid future improvement.

**Comment by kaczynskid-sag** [comment]
If you mean to have it as a separate module within Microservice SDK, than it may be @sag-eweingaertner.
Ultimately it'd be nice if we had some technology agnostic contract, i.e. https://json-schema.org/, that would describe the microservice manifest and allow for it's validation in much simpler ways and possibly without the need for an object mapper.

**Comment by ghost** [comment]
@torsSAG  As @kaczynskid-sag has written, our primary goal currently is to entangle builds with the goal of making CI more stable. Forking the model right now is a sick compromise we have to make in order not to sacrifice Microservice Manifest validation in the SDK.

**Comment by ghost** [comment]
@slc-softwareag: I just saw your PR #159 and figured out that you already did the big (and great thing), namely forking the entire needed model into clients java, including the needed models for the package plugin.

In my opinion, this makes this PR and the entire subtask MTM-46782 obsolete. @kaczynskid-sag @slc-softwareag Should we simply drop this PR?

**Comment by kaczynskid-sag** [comment]
This maven plugin being totally independent of everything does not seem like a bad idea to me. Even if we extract some REST model for the purpose of calling the API, the microservice manifest model is a separate thing. Especially considering the issues with Maven plugins dependencies management I would not couple the SDK model with the plugin model, but rather keep them separate and finish this effort independently. More so because this smaller change alone already unblocks the work on independent SDK and agents pre-merge. Therefore let's leave those two separate.

**Review by bggov — DISMISSED** [review]

**Review by sourajitsaha17 — DISMISSED** [review]

File: microservice/package/src/main/java/com/cumulocity/agent/packaging/PackageMojo.java [hunk 1]
```diff
@@ -356,42 +350,42 @@ private void addPlatformTaggedMicroserviceManifestToZipFile(String targetArchite
...
             ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
             Validator validator = factory.getValidator();
 
-            Stream<ManifestConstraintViolation> violations = validator.validate(manifest).stream().map(input -> new ManifestConstraintViolation(input.getPropertyPath().toString(), input.getMessage()));
+            Set<ConstraintViolation<MicroserviceManifest>> constraintViolations = validator.validate(manifest);
 
-            violations = Stream.concat(violations, validateMemory(manifest).stream());
+            Set<ManifestConstraintViolation> violations = validateMemory(manifest);
+            for (ConstraintViolation<MicroserviceManifest> cviolation: constraintViolations) {
+                violations.add(new ManifestConstraintViolation(cviolation.getPropertyPath().toString(), cviolation.getMessage()));
```
Comment (ghost) [inline microservice/package/src/main/java/com/cumulocity/agent/packaging/PackageMojo.java:371]: From what I see `validateMemory` returns `ImmutableSet`, does `add`ing really works?

**Review by ghost — DISMISSED** [review]

File: microservice/package/pom.xml [hunk 2]
```diff
@@ -43,33 +43,30 @@
 		<maven-plugin-version>3.2.5</maven-plugin-version>
 		<docker-client.version>8.16.0</docker-client.version>
 		<hibernate-validator.version>5.4.3.Final</hibernate-validator.version>
+		<!-- TODO: fix javadoc generation. Will be implemented in DM-1177 -->
+		<maven.javadoc.skip>true</maven.javadoc.skip>
```
Comment (kaczynskid-sag) [inline microservice/package/pom.xml:47]: Good call, thank you :)

**Review by kaczynskid-sag — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Review by kaczynskid-sag — APPROVED** [review]

**Pull request merged**